packages/block-editor/src/store/reducer.js

```diff
@@ -1094,6 +1094,8 @@ function selection( state = {}, action ) {
 
 			return { clientId: blockToSelect.clientId };
 		}
+		case 'RESET_BLOCKS':
+			return {};
```

Comment:
This is a pretty basic solution at the moment and it doesn't cover that there might be cases where selection should be retained. Maybe if the client id for the currently selected block exists in the blocks array from the `RESET_BLOCKS` action?

It looks like `RESET_BLOCKS` is used for synchronizing templates, so that looks like a case where selection could be unexpectedly lost.

However, with multi-selections, trying to retain a selection might cause a situation where `selectionStart` is retained but `selectionEnd` is cleared or vice versa.